src/provider/MetadataStore.ts

```diff
@@ -105,7 +105,7 @@ export class MetadataStore {
      */
     public update(appData: IApplication, app: Application): void {
         if (!this.appData.hasOwnProperty(appData.id)) {
-            this.appData[appData.id] = {uuid: app.identity.uuid, name: app.identity.name!, directoryId: appData.id};
+            this.appData[appData.id] = {uuid: app.identity.uuid, name: app.identity.uuid, directoryId: appData.id};

```

Comment:
Obviously this looks wrong but I assume it's deliberate? Worth a comment here?

Reply:
An appIdentity doesn't have a name. We need a full identity to send it messages. Main window of an app has name===uuid